### ISSUE TITLE
"Huge rendering" issue

### ISSUE THREAD

**Issue by gpasq** [zippyui/react-datagrid#152]
I'm testing out the datagrid in my app and seeing some _funky_ results.

Using react 0.14.2 and the grid is rendering huge sort arrows, and the data is stacked unintelligibly at the bottom.  See attached screenshot.  I've tried removing all of my CSS files but that doesn't change the rendering behavior.

First screen shot shows the rendering of the header.  Second screenshot shows the data jumbled at the bottom of the render.

Package file:
```
{
  "name": "order-management",
  "version": "1.0.1",
  "description": "Order Management app.",
  "keywords": [],
  "main": "index.html",
  "scripts": {
    "start": "gulp clean && gulp",
    "test": "jest"
  },
  "author": "greg.pasq@motili.com",
  "license": "",
  "devDependencies": {
    "babel-core": "^6.1.2",
    "babel-jest": "^5.3.0",
    "babel-loader": "^6.1.0",
    "babel-preset-es2015": "^6.1.2",
    "babel-preset-react": "^6.1.2",
    "del": "^2.0.2",
    "gulp": "^3.9.0",
    "gulp-autoprefixer": "^3.1.0",
    "gulp-concat": "^2.6.0",
    "gulp-connect": "^2.2.0",
    "gulp-livereload": "^3.8.1",
    "gulp-load-plugins": "^1.1.0",
    "gulp-size": "^2.0.0",
    "gulp-stylus": "^2.1.0",
    "gulp-uglify": "^1.4.2",
    "gulp-util": "^3.0.7",
    "gulp-watch": "^4.3.5",
    "gulp-webpack": "^1.5.0",
    "jest-cli": "^0.7.1",
    "object-assign": "^4.0.1",
    "orchestrator": "^0.3.7",
    "react-addons-test-utils": "^0.14.2",
    "webpack": "^1.12.3"
  },
  "dependencies": {
    "history": "^1.13.0",
    "react": "^0.14.2",
    "react-addons-linked-state-mixin": "^0.14.5",
    "react-bootstrap": "^0.28.1",
    "react-datagrid": "^2.0.1",
    "react-dom": "^0.14.2",
    "react-router": "^1.0.0",
    "reflux": "^0.3.0",
    "underscore": "^1.8.3"
  },
  "jest": {
    "scriptPreprocessor": "<rootDir>/node_modules/babel-jest",
    "unmockedModulePathPatterns": [
      "<rootDir>/node_modules/react",
      "<rootDir>/node_modules/react-dom",
      "<rootDir>/node_modules/react-addons-test-utils",
      "<rootDir>/node_modules/fbjs",
      "<rootDir>/node_modules/object-assign"
    ]
  }
}

```

And here is my code; pretty simple:

```
import React from 'react';
import ItemActions from '../actions/itemActions';
import DataGrid from 'react-datagrid';

var WorkOrders = React.createClass({
    data: [{
        number: "50201",
        status: "Tenant Contact Required",
        priority: "General",
        assignedTo: "Troy's HVAC Troy Schumaker",
        ref1: "asdf",
        ref2: "asdf",
        address: "3016 White Oak Lane Highlands Ranch CO 80129",
        created: "12/30/15 4:06 PM"
    }],

    columns: [{
        name: "number",
        title: "Number",
        width: 50
    },
    {
        name: "status",
        title: "Status"
    },
    {
        name: "priority",
        title: "Priority"
    },
    {
        name: "assignedTo",
        title: "Assigned To"
    },
    {
        name: "ref1",
        title: "Ref1"
    },
    {
        name: "ref2",
        title: "Ref2"
    },
    {
        name: "address",
        title: "Address"
    },
    {
        name: "created",
        title: "Created At"
    }],

    render: function() {
        return (
            <div>
                <DataGrid
                    idProperty='id'
                    dataSource={this.data}
                    columns={this.columns}
                    />
            </div>
        );
    }
})

export default WorkOrders;
```

<img width="1245" alt="screen shot 2016-01-05 at 8 23 54 am" src="https://cloud.githubusercontent.com/assets/3801469/12118862/04705eb6-b386-11e5-879c-f6b395593868.png">
<img width="517" alt="screen shot 2016-01-05 at 8 25 53 am" src="https://cloud.githubusercontent.com/assets/3801469/12118864/07e8c27c-b386-11e5-87b2-e709f99caedc.png">

**Comment by radubrehar** [comment]
Have you included the css that comes with the datagrid? this seems to be the problem to me

**Comment by nikhilk78** [comment]
I hope you have included index.css which has styling for this component


Sent from my iPhone

On Jan 5, 2016, at 10:28 AM, Greg Pasquariello <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm testing out the datagrid in my app and seeing some funky results.

Using react 0.14.2 and the grid is rendering huge sort arrows, and the data is stacked unintelligibly at the bottom. See attached screenshot. I've tried removing all of my CSS files but that doesn't change the rendering behavior.

Package file:

{
  "name": "order-management",
  "version": "1.0.1",
  "description": "Order Management app.",
  "keywords": [],
  "main": "index.html",
  "scripts": {
    "start": "gulp clean && gulp",
    "test": "jest"
  },
  "author": "greg.pasq@motili.com<mailto:greg.pasq@motili.com>",
  "license": "",
  "devDependencies": {
    "babel-core": "^6.1.2",
    "babel-jest": "^5.3.0",
    "babel-loader": "^6.1.0",
    "babel-preset-es2015": "^6.1.2",
    "babel-preset-react": "^6.1.2",
    "del": "^2.0.2",
    "gulp": "^3.9.0",
    "gulp-autoprefixer": "^3.1.0",
    "gulp-concat": "^2.6.0",
    "gulp-connect": "^2.2.0",
    "gulp-livereload": "^3.8.1",
    "gulp-load-plugins": "^1.1.0",
    "gulp-size": "^2.0.0",
    "gulp-stylus": "^2.1.0",
    "gulp-uglify": "^1.4.2",
    "gulp-util": "^3.0.7",
    "gulp-watch": "^4.3.5",
    "gulp-webpack": "^1.5.0",
    "jest-cli": "^0.7.1",
    "object-assign": "^4.0.1",
    "orchestrator": "^0.3.7",
    "react-addons-test-utils": "^0.14.2",
    "webpack": "^1.12.3"
  },
  "dependencies": {
    "history": "^1.13.0",
    "react": "^0.14.2",
    "react-addons-linked-state-mixin": "^0.14.5",
    "react-bootstrap": "^0.28.1",
    "react-datagrid": "^2.0.1",
    "react-dom": "^0.14.2",
    "react-router": "^1.0.0",
    "reflux": "^0.3.0",
    "underscore": "^1.8.3"
  },
  "jest": {
    "scriptPreprocessor": "<rootDir>/node_modules/babel-jest",
    "unmockedModulePathPatterns": [
      "<rootDir>/node_modules/react",
      "<rootDir>/node_modules/react-dom",
      "<rootDir>/node_modules/react-addons-test-utils",
      "<rootDir>/node_modules/fbjs",
      "<rootDir>/node_modules/object-assign"
    ]
  }
}



And here is my code; pretty simple:

import React from 'react';
import ItemActions from '../actions/itemActions';
import DataGrid from 'react-datagrid';

var WorkOrders = React.createClass({
    data: [{
        number: "50201",
        status: "Tenant Contact Required",
        priority: "General",
        assignedTo: "Troy's HVAC Troy Schumaker",
        ref1: "asdf",
        ref2: "asdf",
        address: "3016 White Oak Lane Highlands Ranch CO 80129",
        created: "12/30/15 4:06 PM"
    }],

    columns: [{
        name: "number",
        title: "Number",
        width: 50
    },
    {
        name: "status",
        title: "Status"
    },
    {
        name: "priority",
        title: "Priority"
    },
    {
        name: "assignedTo",
        title: "Assigned To"
    },
    {
        name: "ref1",
        title: "Ref1"
    },
    {
        name: "ref2",
        title: "Ref2"
    },
    {
        name: "address",
        title: "Address"
    },
    {
        name: "created",
        title: "Created At"
    }],

    render: function() {
        return (
            <div>
                <DataGrid
                    idProperty='id'
                    dataSource={this.data}
                    columns={this.columns}
                    />
            </div>
        );
    }
})

export default WorkOrders;


[screen shot 2016-01-05 at 8 23 54 am]<https://cloud.githubusercontent.com/assets/3801469/12118862/04705eb6-b386-11e5-879c-f6b395593868.png>
[screen shot 2016-01-05 at 8 25 53 am]<https://cloud.githubusercontent.com/assets/3801469/12118864/07e8c27c-b386-11e5-87b2-e709f99caedc.png>

-

**Comment by gpasq** [comment]
Indeed, looks like that may be the problem.  Having trouble with css-loader.  Grr.

**Comment by radubrehar** [comment]
No problem, let us know how it goes

**Comment by rjeffer** [comment]
Does anyone know why the table doesnt in IE but works in firefox?

**Comment by gpasq** [comment]
Got the webpack issue worked out, looks good.  Thanks for the help, sorry for the bug report!

<img width="1581" alt="screen shot 2016-01-05 at 9 44 35 am" src="https://cloud.githubusercontent.com/assets/3801469/12121171/208cc4da-b391-11e5-9a04-dfdaa580522c.png">

**Comment by rjeffer** [comment]
How did you get your table to display in safari?

**Comment by nikhilk78** [comment]
This is related to flex I think
Try this in css
.react-datagrid > .z-inner {
flex :1 0 auto !imporant
}

Sent from my iPhone

On Jan 5, 2016, at 11:44 AM, rjeffer <notifications@github.com<mailto:notifications@github.com>> wrote:


Does anyone know why the table doesnt in IE but works in firefox?

-

**Comment by gpasq** [comment]
Had no issues in Safari (after the webpack css-loader issue was resolved).  Here's a safari shot:

<img width="1311" alt="screen shot 2016-01-05 at 9 48 41 am" src="https://cloud.githubusercontent.com/assets/3801469/12121273/ad3548d0-b391-11e5-8853-455f8f0078b1.png">

**Comment by rjeffer** [comment]
Thank you @nikhilk78 . It is working fine in IE. I really appreciate it. But for some odd reason, it is giving me issues in Safari. @gpasq you mentioned the web pack loader, can you explain to me what you did?

**Comment by sunilratada** [comment]
Hello,

I am facing same issue and I am new to webpack so can you please guide me how to load CSS file in webpack ?

regards,
Sunil

**Comment by kevinrue** [comment]
Same as @sunilratada 
Glad to see that @gpasq solved it in the end, I am trying to solve it myself.
Can anyone please share a link to a recommended webpack/css-loader tutorial or example?
That would be really appreciated! It's a bit hard climbing the learning curve alone :)

**Comment by kevinrue** [comment]
If that can help debugging, here is my GitHub repository for my app. I have spent full days now trying to figure out how to load the "./node_modules/react-datagrid/index.css" file, but none of them work. I have tried to require relative links like "../node_modules/react-datagrid/index.css" and I've also tried to get webpack to work, but even with tutorial like [this one](http://survivejs.com/webpack/advanced-techniques/configuring-react/) nothing works.

The only way i got the app to display properly was to literally copy-paste the content of react-datagrid index.css inside the main.css file in the client folder. Dirty fix.

Would anyone mind checking out my [GitHub repository for this tiny app](https://github.com/kevinrue/reactJS/tree/master/reactDataGrid) (literally just a page that displays a table, at the moment), and tell me what I'm doing wrong? I'm very new to node JS, so I suspect it may be a stupidly simple thing that I'm missing. 

I would really appreciate it. Thanks!
